web/src/settings_data.ts

```diff
@@ -144,7 +144,13 @@ export function user_can_create_private_streams(): boolean {
 }
 
 export function user_can_create_public_streams(): boolean {
-    return user_has_permission(realm.realm_create_public_stream_policy);
+    if (!current_user.user_id) {
+        return false;
+    }
```

Comment:
Is this a spectator check? Probably worth checking explicitly for `null`/`undefined` for that, not just using falsiness. (One almost never wants to use falsiness in good JS code, because it's imprecise -- we definitely don't mean that a user ID of 0 should somehow be treated specially here, for example)

Reply:
Yes, it is for checking spectators. I checked and user_id is actually 0 for spectators and we use same check at multiple places. Other option can be to check it using page_params.is_spectator.

Reply:
Checking `is_spectator` explicitly is always the better policy, since it's more readable.

Reply:
Ok. I think then we can update it in other places as well to use `is_spectator`.